server/src/main/java/de/uftos/services/GradeService.java

```diff
@@ -157,6 +169,8 @@ public void delete(String id) {
       throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
           "This grade is still associated with a student group.");
     }
+    new ConstraintInstanceDeleter(constraintSignatureRepository, constraintInstanceRepository)
+        .removeAllInstancesWithArgumentValue(new String[] {id});
```

Comment:
This method should probably just forward the input to `deleteGrades()`:
```Java
public void delete(String id) {
    deleteGrades(new String[]{id});
}
```

Reply:
I think you marked the wrong lines. Anyway, I already had planned to remove most of the endpoints to delete a single entity, since they are no longer used. I have now done that.